src/ZKMRStakeRegistry.sol

```diff
@@ -194,6 +232,22 @@ contract ZKMRStakeRegistry is
         return operators[operator].x != 0;
     }
 
+    function _keyInUse(PublicKey memory publicKey)
+        private
+        view
+        returns (bool)
+    {
+        return usedKeys[_pubkeyHash(publicKey)];
+    }
+
+    function _deregisterOperator(address operator) private {
+        totalOperators--;
+        PublicKey memory publicKey = operators[operator];
+        delete operators[operator];
+        delete usedKeys[_pubkeyHash(publicKey)];
```

Comment:
From offline convo, i feel we should keep that info if it's not too much trouble. Reason is we do have persistent state on our own database.